Update MicroProfile Health tests to reflect changes introduced by WFLY-18176

Some MicroProfile Health related tests are failing when run against WildFly, due to the changes introduced by https://issues.redhat.com/browse/WFLY-18176
The following output is generated when runing against WildFly 33.0.1.Final:

```
[ERROR] Failures: 
[ERROR]   DefaultReadinessProcedureHealthTest.testHealthEndpoint:77 3 expectations failed.
JSON path checks doesn't match.
Expected: a collection with size <6>
  Actual: <[{name=boot-errors, status=UP}, {name=deployments-status, status=UP, data={DefaultReadinessProcedureHealthTest.war=OK}}, {name=suspend-state, status=UP, data={value=RUNNING}}, {name=server-state, status=UP, data={value=running}}, {name=started-deployment.DefaultReadinessProcedureHealthTest.war, status=UP}, {name=live, status=UP, data={key=value}}, {name=ready-deployment.DefaultReadinessProcedureHealthTest.war, status=UP}]>

JSON path checks.name doesn't match.
Expected: iterable with items ["deployments-status", "boot-errors", "server-state", "ready-deployment.DefaultReadinessProcedureHealthTest.war", "started-deployment.DefaultReadinessProcedureHealthTest.war", "live"] in any order
  Actual: <[boot-errors, deployments-status, suspend-state, server-state, started-deployment.DefaultReadinessProcedureHealthTest.war, live, ready-deployment.DefaultReadinessProcedureHealthTest.war]>

JSON path checks.data doesn't match.
Expected: a collection with size <6>
  Actual: <[null, {DefaultReadinessProcedureHealthTest.war=OK}, {value=RUNNING}, {value=running}, null, {key=value}, null]>

[ERROR]   DefaultReadinessProcedureHealthTest.testReadinessEndpoint:121 2 expectations failed.
JSON path checks doesn't match.
Expected: a collection with size <4>
  Actual: <[{name=server-state, status=UP, data={value=running}}, {name=boot-errors, status=UP}, {name=deployments-status, status=UP, data={DefaultReadinessProcedureHealthTest.war=OK}}, {name=suspend-state, status=UP, data={value=RUNNING}}, {name=ready-deployment.DefaultReadinessProcedureHealthTest.war, status=UP}]>

JSON path checks.name doesn't match.
Expected: iterable with items ["boot-errors", "server-state", "deployments-status", "ready-deployment.DefaultReadinessProcedureHealthTest.war"] in any order
  Actual: <[server-state, boot-errors, deployments-status, suspend-state, ready-deployment.DefaultReadinessProcedureHealthTest.war]>

[ERROR]   NotReadyHealthCheckTest.delayedConstructorTest:250 1 expectation failed.
JSON path checks.name doesn't match.
Expected: iterable with items ["delayed-readiness", "deployments-status", "boot-errors", "server-state"] in any order
  Actual: <[deployments-status, boot-errors, suspend-state, server-state, delayed-readiness]>

[ERROR]   NotReadyHealthCheckTest.unexpectedHealthChecksTest:302 1 expectation failed.
JSON path checks.name doesn't match.
Expected: iterable with items ["ready-deployment.DelayedLivenessHealthCheck.war", "deployments-status", "boot-errors", "server-state"] in any order
  Actual: <[server-state, suspend-state, boot-errors, deployments-status, ready-deployment.DelayedLivenessHealthCheck.war]>

[INFO] 
[ERROR] Tests run: 72, Failures: 4, Errors: 0, Skipped: 1
``` 

The `master` branch must be updated in order to align the reported failing tests with the WildFly implementation.

--- a/microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/DefaultReadinessProcedureHealthTest.java
+++ b/microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/DefaultReadinessProcedureHealthTest.java
@@ -75,13 +75,13 @@ public class DefaultReadinessProcedureHealthTest {
                 .statusCode(HttpStatus.SC_OK)
                 .contentType(ContentType.JSON)
                 .body("status", is("UP"),
-                        "checks", hasSize(6),
+                        "checks", hasSize(7),
                         "checks.status", hasItems("UP", "UP"),
                         "checks.name",
-                        containsInAnyOrder("deployments-status", "boot-errors", "server-state",
+                        containsInAnyOrder("deployments-status", "boot-errors", "server-state", "suspend-state",
                                 String.format("ready-deployment.%s", ARCHIVE_NAME),
                                 String.format("started-deployment.%s", ARCHIVE_NAME), "live"),
-                        "checks.data", hasSize(6),
+                        "checks.data", hasSize(7),
                         "checks.find{it.name == 'live'}.data.key", is("value"));
     }
 
@@ -119,15 +119,16 @@ public class DefaultReadinessProcedureHealthTest {
         RestAssured.get(HealthUrlProvider.readyEndpoint()).then()
                 .contentType(ContentType.JSON)
                 .body("status", is("UP"),
-                        "checks", hasSize(4),
+                        "checks", hasSize(5),
                         "checks.status", hasItems("UP"),
                         "checks.name",
-                        containsInAnyOrder("boot-errors", "server-state", "deployments-status",
+                        containsInAnyOrder("boot-errors", "server-state", "deployments-status", "suspend-state",
                                 readyDeploymentCheckName),
                         "checks.find{it.name == '" + readyDeploymentCheckName + "'}.data", is(nullValue()),
                         "checks.find{it.name == 'boot-errors'}.data", is(nullValue()),
                         "checks.find{it.name == 'deployments-status'}.data", is(notNullValue()),
-                        "checks.find{it.name == 'server-state'}.data.value", is("running"));
+                        "checks.find{it.name == 'server-state'}.data.value", is("running"),
+                        "checks.find{it.name == 'suspend-state'}.data.value", is("RUNNING"));
     }
 
     /**

--- a/microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/delay/NotReadyHealthCheckTest.java
+++ b/microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/delay/NotReadyHealthCheckTest.java
@@ -41,7 +41,7 @@ import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
 import org.wildfly.extras.creaper.core.online.operations.admin.Administration;
 
 /**
- * Performs tests that aim to verify the MP Health check implementation in complex scenarios where delayed
+ * Performs tests that aim to verify the MP Health check implementation in complex scenarios, where delayed
  * readiness health checks could lead to misinterpretation of server/service status, see
  * https://issues.redhat.com/browse/WFLY-12952.
  * <p>
@@ -249,7 +249,7 @@ public class NotReadyHealthCheckTest {
                         .statusCode(503)
                         .body("status", is("DOWN"),
                                 "checks.name", containsInAnyOrder(DelayedReadinessHealthCheck.NAME, "deployments-status",
-                                        "boot-errors", "server-state"));
+                                        "boot-errors", "server-state", "suspend-state"));
             } finally {
                 controller.stop(ManualTests.ARQUILLIAN_CONTAINER);
             }
@@ -302,7 +302,7 @@ public class NotReadyHealthCheckTest {
                         .body("status", is("UP"),
                                 "checks.name", containsInAnyOrder(
                                         "ready-deployment." + DelayedLivenessHealthCheck.class.getSimpleName() + ".war",
-                                        "deployments-status", "boot-errors", "server-state"));
+                                        "deployments-status", "boot-errors", "server-state", "suspend-state"));
             } finally {
                 controller.stop(ManualTests.ARQUILLIAN_CONTAINER);
             }

--- a/microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/delay/ReadinessChecker.java
+++ b/microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/delay/ReadinessChecker.java
@@ -51,20 +51,20 @@ public class ReadinessChecker implements Callable<Boolean> {
                         if (response.getStatusCode() == 503) {
                             if (checks == null) {
                                 addState(DOWN_NO_CONTENT());
-                            } else if ((checks.size() == 4) && contains(checks, "empty-readiness-checks")) {
+                            } else if ((checks.size() == 5) && contains(checks, "empty-readiness-checks")) {
                                 addState(DOWN_NO_CHECK());
-                            } else if ((checks.size() == 4) && contains(checks, DelayedReadinessHealthCheck.NAME)) {
+                            } else if ((checks.size() == 5) && contains(checks, DelayedReadinessHealthCheck.NAME)) {
                                 addState(ReadinessState.DOWN_WITH_CHECK());
                             }
                         } else if (response.getStatusCode() == 200) {
                             if (checks == null) {
                                 throw new RuntimeException("Readiness probe is UP (200) however missing JSON content");
                             }
-                            if ((checks.size() == 4) && contains(checks, "empty-readiness-checks")) {
+                            if ((checks.size() == 5) && contains(checks, "empty-readiness-checks")) {
                                 addState(UP_NO_CHECK());
-                            } else if ((checks.size() == 4) && contains(checks, DelayedReadinessHealthCheck.NAME)) {
+                            } else if ((checks.size() == 5) && contains(checks, DelayedReadinessHealthCheck.NAME)) {
                                 addState(UP_WITH_CHECK());
-                            } else if ((checks.size() == 4) && contains(checks, DEFAULT_READINESS_CHECK_NAME_PREFIX)) {
+                            } else if ((checks.size() == 5) && contains(checks, DEFAULT_READINESS_CHECK_NAME_PREFIX)) {
                                 addState(UP_WITH_DEFAULT_CHECK());
                             }
                         }

